packages/cli/src/commands/search/dump.ts

```diff
@@ -0,0 +1,169 @@
+import {Command, flags} from '@oclif/command';
+import {AuthenticatedClient} from '../../lib/platform/authenticatedClient';
+import {
+  Preconditions,
+  IsAuthenticated,
+} from '../../lib/decorators/preconditions/';
+import {Config} from '../../lib/config/config';
+import PlatformClient from '@coveord/platform-client';
+import {writeFile} from 'fs-extra';
+import {Parser} from 'json2csv';
+import {cli} from 'cli-ux';
+import {
+  buildAnalyticsFailureHook,
+  buildAnalyticsSuccessHook,
+} from '../../hooks/analytics/analytics';
+
+interface SearchResult {
+  raw: {rowid: string};
+}
+
+interface ProgressBar {
+  update: (up: number) => void;
+  stop: () => void;
+  start: (total: number, current: number) => void;
+  setTotal: (tot: number) => void;
+  increment: (inc: number) => void;
+}
+
+interface SearchResponse {
+  indexToken: string;
+  results: SearchResult[];
+  totalCount: number;
+}
+
+interface FetchParameters {
+  client: PlatformClient;
+  organizationId: string;
+  source: string;
+  additionalFilter: string;
+}
+
+export default class Dump extends Command {
+  static description =
+    'Dump the whole content of a particular source in a CSV format.';
+
+  static flags = {
+    source: flags.string({
+      char: 's',
+      description:
+        'The identifier of the source for which to extract all documents.',
+      helpValue: 'mySourceName',
+      required: true,
```

Comment:
I think that in some cases you will want to extract an aggregate of sources.
Will it be considered too dangerous to make the source flag optional?

Also, I think that a pipeline flag would be really useful.
**A past use case: "Extract all the documents from my Salesforce Agent pipeline."**

Reply:
Also, if the "default" pipeline has filters, the dump will may not contain all the documents.

Reply:
I can make source repeatable.

As for the pipeline, this is a great idea, Ill add it.